lib/vsc/utils/missing.py

```diff
@@ -282,11 +281,20 @@ def __getitem__(self, key, *args, **kwargs):
                 raise KeyError("Unknown key '%s' for %s instance (known keys: %s)" % tup)
 
 
-def shell_quote(x):
-    """Add quotes so it can be apssed to shell"""
-    # use undocumented subprocess API call to quote whitespace (executed with Popen(shell=True))
-    # (see http://stackoverflow.com/questions/4748344/whats-the-reverse-of-shlex-split for alternatives if needed)
-    return subprocess.list2cmdline([str(x)])
+def shell_quote(token):
+    """
+    Wrap provided token in single quotes (to escape space and characters with special meaning in a shell),
+    so it can be used in a shell command.
+    """
+    # make sure we're dealing with a string
+    if not isinstance(token, basestring):
+        token = str(token)
+    # check whether token is already wrapped in single quotes
+    is_escaped = re.compile("^'.*'$")
+    if not is_escaped.match(token):
+        # escape any non-escaped single quotes, and wrap entire token in single quotes
+        token = "'%s'" % re.sub(r"([^\\])'", r"\1\\'", token)
+    return token
```

Comment:
`list2cmdline` doesn't take special characters like `$`, `)`, `(` into account at all, it only cares about whitespace and backslashes; besides, it's a (purposely) undocumented function, so better not to rely on it...

simply quoting everything with single quotes (carefully) is better/safer imho... we want to avoid that the shell does any interpretation at all